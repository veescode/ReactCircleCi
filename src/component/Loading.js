import React from 'react';
import {
  View,
  Text,
  ActivityIndicator,
  StatusBar,
  TouchableWithoutFeedback,
} from 'react-native';

export default ({onCancel}) => (
  <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
    <ActivityIndicator size="large" />
    <StatusBar barStyle="default" />
    <TouchableWithoutFeedback onPress={() => onCancel()}>
      <Text>Cancel</Text>
    </TouchableWithoutFeedback>
  </View>
);
