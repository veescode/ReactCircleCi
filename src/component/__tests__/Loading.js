/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../Loading';

describe('Loader tests', () => {
  const onCancel = jest.fn();
  const instance = renderer.create(<Loading onCancel={onCancel} />);

  it('Should render without any issues', () => {
    expect(instance).toMatchSnapshot();
  });
});
