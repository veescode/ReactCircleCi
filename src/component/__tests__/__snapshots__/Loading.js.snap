// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader tests Should render without any issues 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <Text
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    Cancel
  </Text>
</View>
`;
